### PR TITLE
Use git.k8s.io links for code contributor and fix broken links

### LINKED ISSUE #10578: Link to developer documentation broken
**This is a...** 

- [ ] Feature Request
- [ X] Bug Report

**Problem:**

[Link to developer guide on this page](https://kubernetes.io/docs/home/?path=contributors&persona=code-contributor&level=foundational) broken. 

**Proposed Solution:**

Update link to point to existent documentation.

**Page to Update:**
https://kubernetes.io/docs/community/devel/

### PR DESCRIPTION
**Pull request by shubheksha**
Fixes #10578

### CODE REVIEW

**Comment by tengqm** [comment]
@shubheksha Could you please check if #10591 is a better fix for the problem.

**Comment by shubheksha** [comment]
@tengqm I'm not sure how importing the entire file is better than just linking to it? Could you elaborate on that?

File: data/user-personas/contributors/code-contributor.yaml [hunk 1]
```diff
@@ -5,15 +5,15 @@ index: 0
...
+    url: "https://git.k8s.io/community/keps/"
   - label: "Understand the API conventions"
     icon: fa-map-o
-    url: "https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md"
+    url: "https://git.k8s.io/community/contributors/devel/api-conventions.md"
 advanced:
   - label: "Learn about the release process"
     icon: fa-code-fork
-    url: "https://github.com/kubernetes/release/blob/master/README.md"
+    url: "https://git.k8s.io/release/README.md"
```
Comment (shubheksha) [inline data/user-personas/contributors/code-contributor.yaml]: @tengqm are these files imported too? Or is it just files from the community repo?

Comment (tengqm) [reply data/user-personas/contributors/code-contributor.yaml]: @shubheksha Check this: https://github.com/kubernetes/website/tree/master/update-imported-docs

The difference between direct linking and importing files is that the later is meant to be browsed as web pages while the former is meant for contributors to check the markdown source files.

Comment (tengqm) [reply data/user-personas/contributors/code-contributor.yaml]: As for files to be imported, they can be customized using this file: https://github.com/kubernetes/website/blob/master/update-imported-docs/community.yml

**Comment by Bradamant3** [comment]
@shubheksha and @tengqm the community files are removed by request from sig-contribex. See #10501 . That said, this PR represents a good catch of a 404 -- I'm looking into it and will fix in a separate PR.

**Pull request closed without merging**